docs/database-classes.md

```diff
@@ -72,7 +72,75 @@ public abstract class Person
 }
 ```
 
-**Note**: If a database class definition of a Starcounter application contains any non-computed instance properties that are not declared as abstract auto-implemented properties with public get and set accessors and with one of the [supported data types](database-types.md), an exception will be thrown when the application starts.
+**Note**: If a database class definition of a Starcounter application contains any non-computed instance properties that are not declared as abstract auto-implemented properties with public get and set accessors, an exception will be thrown when the application starts.
+
+## Proxy state fields and properties
+
+Sometimes it is required to declare a public non-persistent field or property in a database class. Such fields or properties can be used for advanced calculations or caching of computed values. This can be achieved with usage of the `ProxyStateAttribute`. Starcounter ignores and does not persist anything marked with the `ProxyStateAttribute`.
+
+```csharp
+[Database]
+public abstract class Person
+{
+    public abstract string Name { get; set; }
+
+    public int NameLength => Name.Length;
+
+    // This property is ignored by Starcounter and is not persistent.
+    [ProxyState]
+    public string CachedInfo { get; set; }
+}
+
+public static IList<Person> SortPeopleByInfo(IList<Person> people)
+{
+    // First, iterate through each person and calculate it's info string.
+    foreach (Person p in people)
+    {
+        // The info string is saved in the non-persistent proxy state property.
+        p.CachedInfo = $"{p.Name} created at {p.CreatedAtUtc} with name length of {p.NameLength}.";
```

Comment:
Still using `p.CreatedAtUtc` here @miyconst, although the property was removed. 🙂